sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HDFSFileSink.java

```diff
@@ -237,7 +239,16 @@ public HDFSWriter(HDFSWriteOperation<K, V> writeOperation,
     }
 
     @Override
-    public void open(String uId) throws Exception {
+    public final void open(String uId,
+                           @Nullable BoundedWindow window,
+                           @Nullable PaneInfo paneInfo,
+                           int shard,
+                           int numShards) throws Exception {
+      if (window != null) {
```

Comment:
Just to be extra safe, in case there's a bug in the caller, might as well `window != null || paneInfo != null`